modules/core/04-channel/types/upgrade.go

```diff
@@ -69,3 +71,22 @@ func (uf UpgradeFields) ValidateBasic() error {
 func (ut UpgradeTimeout) IsValid() bool {
 	return !ut.Height.IsZero() || ut.Timestamp != 0
 }
+
+// HasPassed returns true if the upgrade has passed the timeout height or timestamp
+func (ut UpgradeTimeout) HasPassed(ctx sdk.Context) (bool, error) {
```

Comment:
Could we add some unit tests for this? 

It should be possible to pass a context by either by adding a coordinator and test chain to the `TypesTestSuite` or maybe using the new [testutil pkg func](https://github.com/cosmos/cosmos-sdk/blob/main/testutil/context.go#L20) from the sdk. 

At a glance it looks like it would be easier to use the coordinator to malleate time

Reply:
i was thinking about this method but wasnt sure if it was worth reworking the TypesTestSuite just for this one test. but i can update!